applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_plane_strain_2d.cpp

```diff
@@ -0,0 +1,452 @@
+        rValue = mInelasticFlag;
+    }
+    return rValue;
+}
+/*
+LinearJ2PlasticityPlaneStrain2D::StrainMeasure LinearJ2PlasticityPlaneStrain2D::GetStrainMeasure()
+{
+    return ConstitutiveLaw::StrainMeasure_Infinitesimal;
+}
+
+LinearJ2PlasticityPlaneStrain2D::StressMeasure LinearJ2PlasticityPlaneStrain2D::GetStressMeasure()
+{
+    return ConstitutiveLaw::StressMeasure_Cauchy;
+}
+*/
+
+void LinearJ2PlasticityPlaneStrain2D::InitializeMaterial(const Properties& material_prop,
+                                                                const GeometryType& rElementGeometry,
+                                                                const Vector& rShapeFunctionsValues)
+{
+    mPlasticStrainOld = ZeroVector(this->GetStrainSize());
+    mAccumulatedPlasticStrainOld = 0.;
+}
+
+void LinearJ2PlasticityPlaneStrain2D::FinalizeSolutionStep(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    mPlasticStrainOld = mPlasticStrain;
+    mAccumulatedPlasticStrainOld = mAccumulatedPlasticStrain;
+}
+
+void LinearJ2PlasticityPlaneStrain2D::CalculateMaterialResponsePK1(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+void LinearJ2PlasticityPlaneStrain2D::CalculateMaterialResponsePK2(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+void LinearJ2PlasticityPlaneStrain2D::CalculateMaterialResponseKirchhoff(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+void LinearJ2PlasticityPlaneStrain2D::CalculateMaterialResponseCauchy(Parameters& rValues)
+{
+    const Properties& matprops = rValues.GetMaterialProperties();
+    Vector& epsilon = rValues.GetStrainVector();
+    Vector& StressVector = rValues.GetStressVector();
+    Matrix ElasticityTensor;
+    Matrix& TangentTensor =
+        rValues.GetConstitutiveMatrix(); // TODO find proper getter
+    double hardening_modulus = matprops[ISOTROPIC_HARDENING_MODULUS];
+    double delta_k = matprops[INFINITY_HARDENING_MODULUS];
+    double hardening_exponent = matprops[HARDENING_EXPONENT];
+    double trial_yield_function;
+    double E = matprops[YOUNG_MODULUS];
+    double poisson_ratio = matprops[POISSON_RATIO];
+    double mu = E / (2. + 2. * poisson_ratio);
+    double volumetric_modulus = E / (3 * (1 - 2 * poisson_ratio));
+
+    if (rValues.GetProcessInfo().Has(INITIAL_STRAIN))
+    {
+        noalias(epsilon) += rValues.GetProcessInfo()[INITIAL_STRAIN];
+    }
+
+    mPlasticStrain = mPlasticStrainOld;
+    mAccumulatedPlasticStrain = mAccumulatedPlasticStrainOld;
+
+    ElasticityTensor.resize(4, 4);
+    CalculateElasticityTensor(matprops, ElasticityTensor);
+    Vector sigma_trial;
+    sigma_trial.resize(4);
+    sigma_trial = prod(ElasticityTensor, epsilon - mPlasticStrainOld);
+
+    // StressTrialDev = sigma - 1/3 tr(sigma) * I
+    Vector StressTrialDev;
```

Comment:
Vector StressTrialDev = sigma_trial,

Reply:
Vector stress_trial_dev = sigma_trial
See the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide#local-variables)